### PR TITLE
Update MyLinkedList.java

### PR DESCRIPTION
**Pull request by joej0e**

### CODE REVIEW

File: src/main/java/core/basesyntax/MyLinkedList.java [hunk 1]
```diff
@@ -3,48 +3,160 @@
...
+        last = newNode;
+        if (previous == null) {
+            first = newNode;
+        } else {
+            previous.next = newNode;
+        }
+        size++;
+    }
+
+    private T unlink(Node<T> n) {
```
Comment (boroda4436) [inline src/main/java/core/basesyntax/MyLinkedList.java]: @joej0e this is wondering, but I saw this method in a several people... With the equal implementation

File: src/main/java/core/basesyntax/MyLinkedList.java [hunk 2]
```diff
@@ -3,48 +3,160 @@
...
+            element = last;
+            for (int i = size - 1; i > index; i--) {
+                element = element.previous;
+            }
+        }
+        return element;
+    }
+
+    private static class Node<T> {
+        T item;
```
Comment (boroda4436) [inline src/main/java/core/basesyntax/MyLinkedList.java]: @joej0e access modifier

File: src/main/java/core/basesyntax/MyLinkedList.java [hunk 3]
```diff
@@ -3,48 +3,160 @@
...
+        Node next;
+        Node previous;
+
+        Node(T item, Node previous, Node next) {
```
Comment (boroda4436) [inline src/main/java/core/basesyntax/MyLinkedList.java]:
```suggestion
        Node(T item, Node<T> previous, Node<T> next) {
```

File: src/main/java/core/basesyntax/MyLinkedList.java [hunk 4]
```diff
@@ -3,48 +3,160 @@
...
+    }
+
+    private void checkIndex(int index) {
+        if (index >= size || index < 0) {
+            throw new IndexOutOfBoundsException();
+        }
+    }
+
+    private Node<T> node(int index) {
+        Node element;
```
Comment (boroda4436) [inline src/main/java/core/basesyntax/MyLinkedList.java]:
```suggestion
        Node<T> element;
```

Comment (boroda4436) [reply src/main/java/core/basesyntax/MyLinkedList.java]: in the unlink method you have used parametrized Node.... even have used final keyword)

File: src/main/java/core/basesyntax/MyLinkedList.java (hunk 1 above)
Comment (joej0e) [reply src/main/java/core/basesyntax/MyLinkedList.java]: it happened thanks to the great guru Josh Bloch

File: src/main/java/core/basesyntax/MyLinkedList.java [hunk 5]
```diff
@@ -3,48 +3,160 @@
...
+        return value;
+    }
+
+    private void checkIndex(int index) {
+        if (index >= size || index < 0) {
+            throw new IndexOutOfBoundsException();
+        }
+    }
+
+    private Node<T> node(int index) {
```
Comment (boroda4436) [inline src/main/java/core/basesyntax/MyLinkedList.java]: @joej0e method names should start from verb: getSomethod, addSomethig, doSomething, isSomething, etc...

**Pull request closed without merging**